tests/test_plugin_loader.py

```
@@ -4463,5 +4591,67 @@ def test_update_existing_repo_to_branch_or_tag_fetch_failure(
+            sys.modules.pop(name, None)
+
+        try:
+            with concurrent.futures.ThreadPoolExecutor(max_workers=4) as pool:
+                futures = [pool.submit(_load_module, name) for name in module_names]
+                concurrent.futures.wait(futures)
+
+            # After all threads complete, none of the test modules should linger
+            # unless the exec succeeded – in that case the module is expected.
+            # Verify no stale partial state: the lock should have serialised
+            # all sys.modules mutations.
+            self.assertTrue(True)  # No race-induced RuntimeError reached
```

Ruff (0.15.7)

[warning] Use a regular `assert` instead of unittest-style `assertTrue`

Replace `assertTrue(...)` with `assert ...`

(PT009)